passes/cmds/publish.cc

```diff
@@ -0,0 +1,40 @@
+#include "kernel/register.h"
+#include "kernel/rtlil.h"
+#include "kernel/log.h"
+
+USING_YOSYS_NAMESPACE
+PRIVATE_NAMESPACE_BEGIN
+
+struct PublishPass : public Pass {
+private:
+	static void publish(RTLIL::IdString& id) {
+		if (id.begins_with("$")) {
+			log_debug("publishing %s\n", id.c_str());
+			id = "\\" + id.str();
+			log_debug("published %s\n", id.c_str());
+		}
+	}
+public:
+	PublishPass() : Pass("publish", "publish private cell types") { }
```

Comment:
I'd say this is a general name for a niche application. What about `chtype -publish_icells`?